File: Chain of responsability/Java/Middle.java
public class Middle implements InterfaceAyuda {
    
    private final int TIPO_AYUDA = 2;
    private InterfaceAyuda sucesor;

    Middle(InterfaceAyuda s)
    {
        this.sucesor = s;
    }
    @Override
    public void getAyuda(int tipo)
    {
        if(tipo == TIPO_AYUDA)
        {
            System.out.println("Ayuda desde el Middle");
        }else{
            this.sucesor.getAyuda(tipo);
        }
    }
}

File: Chain of responsability/Python/Middle.py
from InterfaceAyuda import *

class Middle(InterfaceAyuda):
    
    _TIPO_AYUDA = 2

    def __init__(self, s):
        self._sucesor = s

    def getAyuda(self, tipo):
        if(tipo == self._TIPO_AYUDA):
            print("Ayuda desde el Middle")
        else:
            self._sucesor.getAyuda(tipo)        